js/mail.js

```diff
@@ -1194,6 +1188,35 @@ var Mail = {
 $(document).ready(function() {
 	Mail.UI.initializeInterface();
 
+	// Enables infinite scroll ability.
+	setInterval(function() {
+
+		// Element identifies the element we want to trigger our loading on.
+		//var element = $('.mail_message_summary:nth-last-child(1)');
+		var element = $('#load-more-mail-messages');
+		var outerHeight = element.outerHeight();
+		var offset = element.offset();
+		var scrollLength = $('#mail_messages').scrollTop();
+		var screenHeight = $(document).height();
+		var messageListHeight = scrollLength + offset.top + outerHeight;
+
+		if ((scrollLength + screenHeight) >= messageListHeight) {
+			Mail.UI.messageView.loadMessages('addMore');
+		}
+
+	}, 1000);
+
+	/**
+	 * Checks and loads any new messages and changes to the current folder.
+	 * @todo combine this with the checkForNotifications method.
+	 */
+	setInterval((function() {
+		// Begin the loop.
+		return function() {
+			Mail.UI.messageView.loadMessages();
+		};
+	})(), 10000*1000);
```

Comment:
Good catch. And I need to reset that refresh rate back down to 10 seconds instead of 10000. ;) I was running error tests and needed to stop the interval, so I temporarily changed the timer.